packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -101,6 +101,10 @@ export class FluidSerializer implements IFluidSerializer {
                 ? value.url
                 : generateHandleContextPath(value.url, this.context);
 
+            // Update the root context of all handles that are decoded.
+            // TODO: The cast to any is done temporarily until the change to IFluidHandleContext in `core-interfaces` is
+            // merged and client is updated to the new `core-interfaces` version.
+            (this.root as any).handleDecoded?.(absolutePath);
```

Comment:
I think you are saying (and maybe I misunderstood comment initially) that you will break this PR into two, and it will be done properly, right? If so, it's better, but I think we need to still figure out next step - how to make sure this is not optional (in 2 releases), so comment and follow up issues should track it

Reply:
Yep, I will break this PR into two once these changes look good.